features/login/impl/src/main/kotlin/io/element/android/features/login/impl/login/LoginFlowView.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.features.login.impl.login
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import io.element.android.features.login.impl.R
+import io.element.android.features.login.impl.dialogs.SlidingSyncNotSupportedDialog
+import io.element.android.features.login.impl.error.ChangeServerError
+import io.element.android.features.login.impl.screens.confirmaccountprovider.LoginFlow
+import io.element.android.features.login.impl.screens.createaccount.AccountCreationNotSupported
+import io.element.android.libraries.architecture.AsyncData
+import io.element.android.libraries.designsystem.components.dialogs.ErrorDialog
+import io.element.android.libraries.designsystem.theme.LocalBuildMeta
+import io.element.android.libraries.matrix.api.auth.OidcDetails
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@Composable
+fun LoginFlowView(
```

Comment:
I know it's named like this because it's based on the `LoginFlow` value, but it kind of conflicts with the `FlowNode` components we have, so it took me a while to understand this is not managing a flow of screen/components. Maybe we should try thinking about a better name...

Reply:
Maybe rename `LoginFlow` to `AccountSetup` so you have:

```kotlin
sealed interface AccountSetup {
    data object PasswordLogin : AccountSetup
    data class OidcLogin(val oidcDetails: OidcDetails) : AccountSetup
    data class AccountCreation(val url: String) : AccountSetup
}
```

And this can now be `AccountSetupView`. I can't come up with a better name, it's tricky one 🫤 .

Reply:
I think I will go for `LoginMode` and extract this interface.

Reply:
Ok, although `AccountCreation` is *technically* not a login mode 🤓 .